contents/node/nodejs.md

```diff
@@ -62,31 +62,43 @@ Server running at http://127.0.0.1:3000/
 ```
 To test the server, open a browser tab and navigate to http://localhost:3000/. You should see 'Hello world'.
 
+</box>
+
 Read on to find out about how to incorporate external dependencies, manage them and use Node's module system to organize your code.
 
 ### Benefit 2: Avoid Synchronization Problems and Overheads
 
-Node is designed to be **event-driven**. When an event happens e.g. an I/O event is complete, the event handler/callback function will be enqueued in a queue where it will be subsequently scheduled to run by the [event loop](https://medium.com/the-node-js-collection/what-you-should-know-to-really-understand-the-node-js-event-loop-and-its-metrics-c4907b19da4c). The main event loop (where Javascript code is executed) is single-threaded.
+<figure>
+<img src="eventloop.png" width="74%" class="mx-auto mt-3 mb-2 d-block rounded" />
+<figcaption class="text-center"><small>
+
+Illustration of nodeJS's event loop from [these slides](https://www.slideshare.net/DevCentralAMD/introduction-to-nodejs-41469620)</small>
+</figcaption>
+</figure>
+
+Node is designed to be **event-driven**. When an <tooltip content="e.g. when a I/O operation is complete, or a timer fires">event</tooltip> occurs, the event handler previously registered with the event is queued to be run by the [event loop](https://medium.com/the-node-js-collection/what-you-should-know-to-really-understand-the-node-js-event-loop-and-its-metrics-c4907b19da4c). The main event loop (where Javascript code is executed) is single-threaded.
 
 This means that we can avoid thread overheads and synchronization problems including deadlocks and race conditions.
 
-### Benefit 3: Fast for I/O Intensive Tasks
+### Benefit 3: Fast for I/O Intensive Programs
+
+I/O requests made in other languages such as Python or Java are typically blocking, which means the program remains idle (in the same thread) until the I/O operation completes.
 
-This is due to the **non-blocking** I/O model.
+In contrast, nodeJS's allows for a **non-blocking** I/O model with its event-driven structure.
 
-When file and other I/O operations requests are made in Python or Java, they are blocking -- subsequent requests (in the same thread) have to be enqueued and can only be processed after the current operation and request completes. Meanwhile, the program remains idle. In contrast, Node has an event loop which delegates the actual tasks to other systems (e.g. file system and databases) and it is almost never blocked after delegation. While an I/O operation is incomplete, the event loop can still process subsequent requests.
+I/O requests are delegated to other systems (e.g. file system and databases). While an I/O operation is incomplete, the event loop can still process subsequent requests. When the I/O request is complete, the handler registered with the request is scheduled to be executed on the event loop again.
```

Comment:
```suggestion
I/O requests are delegated to other systems (e.g. file system and databases). While an I/O operation is incomplete, the event loop can still process subsequent requests. When the I/O request is complete, the handler registered with the request is then scheduled to be executed on the event loop.
```

It's not registered again right? Only when when the operation has completed will the handler be scheduled for the first time.